Content.Shared/Atmos/EntitySystems/SharedGasTileOverlaySystem.cs

```diff
@@ -36,6 +58,29 @@ public override void Initialize()
             VisibleGasId = visibleGases.ToArray();
         }
 
+        private void UpdateMaxHeat(float val)
+        {
+            _tempAtMaxHeatDistortion = val;
+            UpdateHeatSlopeAndIntercept();
+        }
+
+        private void UpdateMinHeat(float val)
+        {
+            _tempAtMinHeatDistortion = val;
+            UpdateHeatSlopeAndIntercept();
+        }
+
+        private void UpdateHeatSlopeAndIntercept()
+        {
+            // Make sure to avoid invalid settings (min == max or min > max)
+            // I'm not sure if CVars can have constraints or if CVar subscribers can reject changes.
+            var diff = _tempAtMinHeatDistortion < _tempAtMaxHeatDistortion
+                ? _tempAtMaxHeatDistortion - _tempAtMinHeatDistortion
+                : 0.001f;
+            _heatDistortionSlope = 1.0f / diff;
+            _heatDistortionIntercept = -_tempAtMinHeatDistortion * _heatDistortionSlope;
+        }
```

Comment:
I don't think this calculation is expensive enough to be worth caching and re-using. It makes the code more confusing, adds more opportunities for missing an update, and makes `GetHeatDistortionStrength()` more difficult to test.